cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -671,4 +707,34 @@ protected static void countStageMetrics(StageMetrics stageMetrics,
       return sqlEngine.transform(sqlContext);
     }));
   }
+
+  /**
+   * Try to write the output directly to the SQLEngineOutput registered by this engine.
+   * @param datasetName dataset to write
+   * @param sqlEngineOutput output instance created by this engine
+   * @return {@link SQLEngineJob<Boolean>} representing if the write operation succeded.
+   */
+  public SQLEngineJob<Boolean> write(String datasetName, SQLEngineOutput sqlEngineOutput) {
+    return runJob(datasetName, SQLEngineJobType.WRITE, () -> {
```

Comment:
I think this can be a problem is we will write a single result into multiple sinks. Could you use e.g. Sink node id as dataset name?

Reply:
You are right, I had not considered this. I'll use the Sink's name as this identifier.